src/waveform/waveform.cpp

```diff
@@ -193,21 +193,21 @@ void Waveform::readByteArray(const QByteArray& data) {
         m_data[i].filtered.high = use_high ? static_cast<unsigned char>(high.value(i)) : 0;
     }
     m_completion = dataSize;
-    m_bDirty = false;
+    m_saveState = SaveState::Saved;
 }
 
 void Waveform::resize(int size) {
     m_dataSize = size;
     m_textureStride = computeTextureStride(size);
     m_data.resize(m_textureStride * m_textureStride);
-    m_bDirty = true;
+    m_saveState = SaveState::SavePending;
```

Comment:
Can we enforce it by checking for the SaveState::SavePending state, before write access the waveform again?